MNTOR-2069-README (#3332)
mozilla/blurts-server: src/e2e/README.md

# How to Run e2e Test

#### Install npm packages

###### Error will show if node 18.12.0^ is not installed --- [volta.sh](https://volta.sh/) is the recommended tool for this situation.

```sh
npm install
```

#### Install playwright along with necessary browser agents

```sh
npx playwright install
```

#### Generate .env from .env-dist

```sh
cp .env-dist .env
```

#### Add some additional values to the .env file

> `E2E_TEST_ENV=` <'local' | 'dev' | 'stage'> stage is default, for prod please use GA

Reviewer: I think you might want to move the backtick for clearer formatting?

```suggestion
`E2E_TEST_ENV=<'local' | 'dev' | 'stage'>` stage is default, for prod please use GA
```

# How to Run e2e Test

> ###

Reviewer 2: ✂️

#### Install npm packages error will show if node 18.12.0^ is not installed, `nvm install 18.12.0` should fix this.

```sh
npm install

> ```

Author: updating

#### Install playwright along with necessary browser agents

```sh
npx playwright install
```

#### Generate .env from .env-dist

```sh
cp .env-dist .env
```

#### Add some additional values to the .env file

`E2E_TEST_ACCOUNT_EMAIL=`

`E2E_TEST_ACCOUNT_PASSWORD=`

> `E2E_TEST_BASE_URL=`

Reviewer 2: might need a comment on what value we expect here. I assume a FQDN, but not sure about trailing slashes, etc.

I was able to search and find a couple examples:
- https://github.com/mozilla/blurts-server/blob/fe9cafba924c924deb1c1146ce08b33d0916f228/src/e2e/utils/helpers.js#L22
- https://github.com/mozilla/blurts-server/blob/fe9cafba924c924deb1c1146ce08b33d0916f228/playwright.config.js#L57
- https://github.com/mozilla/blurts-server/blob/fe9cafba924c924deb1c1146ce08b33d0916f228/src/e2e/utils/helpers.js#L13-L17

Author: will just remove this, as choosing the E2E_TEST_ENV will update this env var

> `E2E_TEST_ENV=` if this is not set, stage will be used

Reviewer 2: I don't see this key in the current .env-dist file, we might want to add it and possibly default it to something (or leave a comment on expected/valid values):

https://github.com/mozilla/blurts-server/blob/fe9cafba924c924deb1c1146ce08b33d0916f228/.env-dist#L131-L134

I was able to get a rough hint by searching for keys in the repo and finding this, but not sure what the allowed values are:
https://github.com/mozilla/blurts-server/blob/fe9cafba924c924deb1c1146ce08b33d0916f228/.github/workflows/playwright.yml#L78

Author: good point, updating the .env-dist to include this

#### Verify helpers.js to make sure correct urls are being used

> #### e2e runs on localhost:6060 by default, inorder to run against other envs, provide the end in command

Reviewer 2: ```suggestion
#### e2e runs on localhost:6060 by default, in order to run against other envs, provide the env in command
```

> `Update the E2E_TEST_ENV in .env file`

Reviewer 2: ```suggestion
Update the `E2E_TEST_ENV` in the .env file
```
